include/zephyr/logging/log_backend_net.h

```diff
@@ -27,6 +28,20 @@ extern "C" {
  */
 bool log_backend_net_set_addr(const char *addr);
 
+/**
+ * @brief Allows user to set a server IP address at runtime
```

Comment:
I think that the brief description should differentiate the two functions,  something like
> Allows user to set a server IP address, provided as sockaddr structure, at runtime

And same for the original one
> Allows user to set a server IP address, provided as string, at runtime